python/tskit/util.py

```diff
@@ -183,6 +183,31 @@ def unpack_arrays(packed, offset):
     return ret
 
 
+def subset_ragged_col(packed, offset, subset, col="metadata"):
```

Comment:
This is a nice, elegant way of doing what we need. I wonder if should be public, though? Is it too specific to our use case? If it doesn't need to be public, we could call it `_subset_ragged_col`  (and then we wouldn't have to test it). I am honestly not sure. 

I do think the argument `col` should be called `type`, and the method should be called `subset_ragged_column`.